cpu/esp32/periph/gpio.c

```diff
@@ -84,17 +87,78 @@ const char* _gpio_pin_usage_str[] =
 
 #ifdef ESP_PM_WUP_PINS
 /* for saving the pullup/pulldown settings of wakeup pins in deep sleep mode */
-static bool _gpio_pin_pu[GPIO_PIN_NUMOF] = { };
-static bool _gpio_pin_pd[GPIO_PIN_NUMOF] = { };
+bool _gpio_pin_pu[GPIO_PIN_NUMOF] = { };
+bool _gpio_pin_pd[GPIO_PIN_NUMOF] = { };
 #endif
 
 #if defined(CPU_FAM_ESP32)
 
-#define GPIO_IN_GET(b)  (b < 32) ? GPIO.in & BIT(b) : GPIO.in1.val & BIT(b-32)
-#define GPIO_OUT_SET(b) if (b < 32) { GPIO.out_w1ts = BIT(b); } else { GPIO.out1_w1ts.val = BIT(b-32); }
-#define GPIO_OUT_CLR(b) if (b < 32) { GPIO.out_w1tc = BIT(b); } else { GPIO.out1_w1tc.val = BIT(b-32); }
-#define GPIO_OUT_XOR(b) if (b < 32) { GPIO.out ^=  BIT(b); } else { GPIO.out1.val ^=  BIT(b-32); }
-#define GPIO_OUT_GET(b) (b < 32) ? (GPIO.out >> b) & 1 : (GPIO.out1.val >> (b-32)) & 1
+static inline int _gpio_reg_in_get(gpio_t pin)
+{
+    return (pin < 32) ? (GPIO.in >> pin) & 1 : (GPIO.in1.val >> (pin-32)) & 1;
```

Comment:
```suggestion
    return (pin < 32) ? (GPIO.in >> pin) & 1 : (GPIO.in1.val >> (pin - 32)) & 1;
```

Reply:
BTW, changing the macros to static inline functions is not needed for this PR. Should I split the PR?

Reply:
It just blows up the PR unecessarily.

Reply:
To me, either is fine